src/v/pandaproxy/rest/handlers.cc

```diff
@@ -470,6 +505,14 @@ post_consumer_offsets(server::request_t rq, server::reply_t rp) {
                           rq.req->content.data(),
                           ppj::partition_offsets_request_handler()));
 
+    std::any_of(req_data.begin(), req_data.end(), [](auto&& r) -> bool {
+        validate_no_control(
+          r.name(),
+          pp_parsing_error{
+            "requested topic name contains invalid control character"});
+        return false;
+    });
+
```

Comment:
Should these go on all user inputs of string type? There are other PP handlers that take user input as strings:
- `get_topics_records`
- `post_topics_name`
- `remove_consumer`
- `consumer_fetch`
- `get_consumer_offsets`
- `post_consumer_offsets`

Reply:
So I'm only checking what gets parsed out of JSON since JSON string values can carry the control characters.  I believe (and maybe I'm wrong) that in, say for example `get_topics_records`, the `topic_name` parameter is actually carried in the URL query as a parameter and not in JSON.  But now that you mention it, I wonder it's still possible to provide a newline as a parameter by passing `%0A` in the parameter value (like `/topics/test%0Aname/partitions/1/records`)

Reply:
Yeah can confirm, if you put `%0a` within the URL parameters it will translate to a new line